src/ooxml/java/org/apache/poi/xwpf/usermodel/XWPFChart.java

```diff
@@ -105,19 +137,219 @@ public boolean equals(Object obj) {
         if (obj == this) {
             return true;
         }
-
+        
         if (obj == null) {
             return false;
         }
-
+        
         if (!(obj instanceof XWPFChart)) {
             return false;
         }
         return false;
     }
-
+    
     @Override
     public int hashCode() {
         return getChecksum().hashCode();
     }
+    
+    /**
+     * method to create relationship with embedded part for example writing xlsx
+     * file stream into output stream
+     * 
+     * @param chartRelation chart relationship object which used to create relation with chart
+     * @param index its chart index
+     * @return return relation part which used to write relation in .rels file
+     *         and get relation id
+     * @since POI 4.0.0
+     */
+    public RelationPart createRelationship(XWPFRelation chartRelation, int index) {
+        return createRelationship(chartRelation, XWPFFactory.getInstance(), index, false);
+    }
+    
+    /**
+     * protected method which used to initialization of sheet output stream
+     * 
+     * @param sheet output stream of embedded xlsx file
+     * @since POI 4.0.0
+     */
+    protected void addEmbeddedWorkSheet(OutputStream sheet) {
+        this.sheet = sheet;
+    }
+    
+    /**
+     * this method is used to write workbook object in embedded part of chart
+     * return true in case of successfully write work book in embedded part or
+     * return false
+     * 
+     * @param wb this is workbook object which we used to write in embedded xlsx file
+     * @return returns true in case of successfully write workbook in embedded
+     *         part or returns false
+     * @throws IOException
+     * @since POI 4.0.0
+     */
+    public boolean writeEmbeddedWorkSheet(Workbook wb) throws IOException {
```

Comment:
yes sir we can do it for both XSLFChart and XWPFChart object in one common class.
but as for embedded part(.xlsx file) when we are using XSSFWorkbook.class in XSLF relation variable or in XWPFRelation.CHART_SHEET and calling saveWorkBook(XSSFWorkbook wb) method at that time it working fine. i mean we are able to write workbook to output stream.but when we are going to write final document object it was trying to write internal part of excel file eg. /doc/app.xml,/xl/workbook.xml etc. and due to that the generated document is corrupted, because of blank excel embedded part. While i am using XWPFChartData.class in relation and calling saveWorkbook method then it;s working fine. we need to look this issue.